pkg/storage/unified/search/bleve.go

```diff
@@ -481,12 +481,22 @@ var textSortFields = map[string]string{
 }
 
 // Convert a "requirement" into a bleve query
-func requirementQuery(req *resource.Requirement, prefix string) (query.Query, *resource.ErrorResult) {
+func requirementQuery(req *resource.Requirement, prefix string, opts *resource.ListOptions) (query.Query, *resource.ErrorResult) {
 	switch selection.Operator(req.Operator) {
 	case selection.Equals, selection.DoubleEquals:
 		if len(req.Values) == 0 {
 			return query.NewMatchAllQuery(), nil
 		}
+
+		if req.Key == resource.SEARCH_FIELD_ID {
+			values := make([]string, 0, len(req.Values))
+			for _, v := range req.Values {
+				id := fmt.Sprintf("%s/%s/%s/%s", opts.Key.Namespace, opts.Key.Group, opts.Key.Resource, v)
```

Comment:
i don't like this replacement here -- should we do the replacement in search.go? or use the name field directly?

Reply:
name field isn't indexed.  i assumed this would be the key for all resource types

Reply:
🤔  I think we will want/need name indexed -- given that the index is limited to a single namespace/group/resource, it *could* be the primary key.   BUT then I am not sure what happens when we federate across multiple resources -- there can (and will) be duplicate names across indexes